keyboards/sofle/keymaps/helltm/keymap.c

```diff
@@ -0,0 +1,759 @@
+            0x04, 0x04, 0x03, 0x01, 0x00, 0x00, 0x09, 0x01, 0x80, 0x80, 0xab, 0x04, 0xf8, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0x1c, 0x20, 0x20, 0x3c, 0x0f, 0x11, 0x1f, 0x02, 0x06,
+            0x18, 0x20, 0x20, 0x38, 0x08, 0x10, 0x18, 0x04, 0x04, 0x02, 0x02, 0x01, 0x00, 0x00, 0x00, 0x00,
+        },
+
+        /* 'sneak2', 32x22px */
+        {
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x80, 0x40, 0x40, 0x40, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0xe0, 0xa0, 0x20, 0x40, 0x80, 0xc0, 0x20, 0x40, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x3e, 0x41, 0xf0, 0x04, 0x02, 0x02, 0x02, 0x03, 0x02, 0x02, 0x02, 0x04,
+            0x04, 0x02, 0x01, 0x00, 0x00, 0x00, 0x04, 0x00, 0x40, 0x40, 0x55, 0x82, 0x7c, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3f, 0x20, 0x30, 0x0c, 0x02, 0x05, 0x09, 0x12, 0x1e, 0x04,
+            0x18, 0x10, 0x08, 0x10, 0x20, 0x28, 0x34, 0x06, 0x02, 0x01, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00,
+        }
+    };
+
+    /* animation */
+    void animate_luna(void) {
+
+        /* jump */
+        if (isJumping || !showedJump) {
+
+            /* clear */
+            oled_set_cursor(LUNA_X,LUNA_Y +2);
+            oled_write("     ", false);
+
+            oled_set_cursor(LUNA_X,LUNA_Y -1);
+
+            showedJump = true;
+        } else {
+
+            /* clear */
+            oled_set_cursor(LUNA_X,LUNA_Y -1);
+            oled_write("     ", false);
+
+            oled_set_cursor(LUNA_X,LUNA_Y);
+        }
+
+        /* switch frame */
+        current_frame = (current_frame + 1) % 2;
+
+        /* current status */
+        if(led_usb_state.caps_lock) {
+            oled_write_raw_P(bark[abs(1 - current_frame)], ANIM_SIZE);
+
+        } else if(isSneaking) {
+            oled_write_raw_P(sneak[abs(1 - current_frame)], ANIM_SIZE);
+
+        } else if(current_wpm <= MIN_WALK_SPEED) {
+            oled_write_raw_P(sit[abs(1 - current_frame)], ANIM_SIZE);
+
+        } else if(current_wpm <= MIN_RUN_SPEED) {
+            oled_write_raw_P(walk[abs(1 - current_frame)], ANIM_SIZE);
+
+        } else {
+            oled_write_raw_P(run[abs(1 - current_frame)], ANIM_SIZE);
+        }
+    }
+
+    /* animation timer */
+    if(timer_elapsed32(anim_timer) > ANIM_FRAME_DURATION) {
+        anim_timer = timer_read32();
+        animate_luna();
+    }
+
+    /* this fixes the screen on and off bug */
+    if (current_wpm > 0) {
+        oled_on();
+        anim_sleep = timer_read32();
+    } else if(timer_elapsed32(anim_sleep) > OLED_TIMEOUT) {
+        oled_off();
+    }
+
+}
+
+/* KEYBOARD PET END */
+
+static void print_logo_narrow(void) {
+    render_logo();
+
+    /* wpm counter */
+    char wpm_str[8];
+    oled_set_cursor(0,14);
+    sprintf(wpm_str, " %03d", current_wpm);
+    oled_write(wpm_str, false);
+
+    oled_set_cursor(0,15);
+    oled_write(" wpm", false);
```

Comment:
If you handle this manually, you will shave 1.5k off the firmware. 


```suggestion
    uint8_t n = get_current_wpm();
    char wpm_str[4];
    oled_set_cursor(0,14);
    wpm_str[3] = '\0';
    wpm_str[2] = '0' + n % 10;
    wpm_str[1] = '0' + ( n /= 10) % 10;
    wpm_str[0] = '0' + n / 10;
    oled_write(wpm_str, false);

    oled_set_cursor(0,15);
    oled_write(" wpm", false);
```

Reply:
interesting! a bit harder to read but surely whoever messes around with qmk is familiar with code, right?

Reply:
well, this is from sigprof initially, but can verify it works, and how it works.  And it saves you 1.5kb, which is not insignificant.  So worth passing on, at least!